### PR TITLE
Readme troubleshooting section

### PR DESCRIPTION
**Pull request by ppalucki**
Also always use cri-resmgr-agent instead 'node agent'

Check the document here: https://github.com/intel/gardener-extension-cri-resmgr/blob/readme-troubleshooting-section/README.md

### CODE REVIEW

**Review by mmucek95 — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -9,6 +9,12 @@
 
 This Gardener extension will deploy and manage lifecycle of [CRI-Resource-Manager](https://github.com/intel/cri-resource-manager) in "shoot" clusters worker nodes deployed by Gardener. **CRI-Resource-Manager** is a proxy between kubectl and containerd to allow more sophisticated node resource management.
 
+### Requirments
```
[the remark this answers is not in the document]
Comment (ppalucki) [reply README.md:12]: done

File: README.md [hunk 2]
```diff
@@ -386,8 +392,60 @@ Unit cri-resource-manager.service could not be found.
...
+1. `ControllerInstallation/cri-resmgr-extension` Valid/Installed/Healthy conditions are true - those conditions are based on `ManagedResource/cri-resmgr-extension-*` (synced by **gardenlet/controllerinstallation_care controller**),
+1. `ManagedResource/cri-resmgr-extension-*` of `type=seed` results in deploying `Pod/cri-resmgr-extension` in seed cluster (**reconciled by gardener-resource-manager**),
+1. (Optionally) if configs are defined, the additional `ConfigMap/gardener-extension-cri-resmgr-configs` should be created in `extension-cri-resmgr-extension-XXXX` namespace with keys representing different kind of configs (e.g. fallback/default),
+1. At the end of shoot reconciliation process, **gardenlet** creates `Extension/cri-resource-manager` watched by `Pod/cri-resmgr-extension` which initiates deployment of CRI-resource-manager to shoot worker nodes,
+1. CRI-Resource-Manager deployment is implemented by creating `ManagedResource/extension-runtime-cri-resmgr` for shoot that consists installation chart (agent and installation DaemonSets and optional `ConfigMap/cri-resmgr-config.TYPE` configs),
```
[the remark this answers is not in the document]
Comment (ppalucki) [reply README.md:406]: done

File: README.md [hunk 3]
```diff
@@ -386,8 +392,60 @@ Unit cri-resource-manager.service could not be found.
...
+1. After **gardener-resource-manager** reconciles `ManagedResources/extension-runtime-cri-resmgr` for shoot - conditions of those newly created resources (2 x DaemonsSets with liveness probes) are monitored by `Pod/cri-resmgr-extension` and reported in `Extension/cri-resmgr-extension` status( `SystemComponentsHealthy` condition),
+1. Condition of `Extensions/cri-resmgr-extension` is monitored by **gardenlet/shoot_care** controller and affects finally `Shoot.status.conditions`
+
+In a shoot, we expect two DaemonSets to be deployed:
+- **cri-resmgr-agent** - that Ready status is based on probl that tries to connect to local CRI-Resource-Manager process,
```
[the remark this answers is not in the document]
Comment (ppalucki) [reply README.md:411]: yes - fixed you're right

File: README.md [hunk 4]
```diff
@@ -386,8 +392,60 @@ Unit cri-resource-manager.service could not be found.
...
+- **cri-resmgr-installation** - which copies binaries, configures CRI-resource-manager and then reconfigures and restarts kubelet and then goes to sleep - its Ready status is based on probe that checks "systemctl status cri-resource-manager"
+
+To debug issues you can consult use logs of following components (commands are for local kind-based Gardener deployment):
```
[the remark this answers is not in the document]
Comment (ppalucki) [reply README.md:414]: of course - fixed

File: README.md [hunk 5]
```diff
@@ -386,8 +392,60 @@ Unit cri-resource-manager.service could not be found.
              --container-runtime-endpoint=unix:///run/containerd/containerd.sock  # <---
 ```
 
-
-### III. Running e2e tests
+## Troubleshooting and debugging issues
+
+### Deployment and propagation of changes
+
+In garden virtual and seed clusters you should monitoring and watch over following process:
```
[the remark this answers is not in the document]
Comment (ppalucki) [reply README.md:399]: done

**Pull request merged**